app/models/tournament.rb

```diff
@@ -21,4 +21,6 @@
 
 class Tournament < Event
   enum game_modes: [:ko, :ko_group, :double_elimination]
+
+  validates :deadline, :startdate, :enddate, presence: true
```

Comment:
good job :) but add this to `league.rb` aswell please, if you end up removing it from `event.rb`